rotkehlchen/data_import/importers/binance.py

```diff
@@ -315,40 +313,20 @@ def process_trades(
                     )
                 continue
 
-            rate = to_amount / from_amount
-            trade = Trade(
-                timestamp=timestamp,
+            swap_events.extend(create_swap_events(   # any row works, just using the first to create the event identifier  # noqa: E501
+                timestamp=ts_sec_to_ms(timestamp),
                 location=Location.BINANCE,
-                trade_type=trade_type,
-                base_asset=to_asset,
-                quote_asset=from_asset,
-                amount=to_amount,
-                rate=Price(rate),
-                fee_currency=fee_asset,
-                fee=fee_amount,
-                link='',
-                notes='Imported from binance CSV file. Binance operation: Buy / Sell',
-            )
-            raw_trades.append(trade)
-
-        # Sometimes we can get absolutely identical trades (including timestamp) but the database
-        # allows us to add only one of them. So we combine these trades into a huge single trade
-        # First step: group trades
-        grouped_trades: dict[TradeID, list[Trade]] = defaultdict(list)
-        for trade in raw_trades:
-            grouped_trades[trade.identifier].append(trade)
-
-        # Second step: combine them
-        unique_trades = []
-        for trades_group in grouped_trades.values():
-            result_trade = trades_group[0]
-            for trade in trades_group[1:]:
-                result_trade.amount += trade.amount
-                if result_trade.fee is not None and trade.fee is not None:
-                    result_trade.fee = Fee(result_trade.fee + trade.fee)
-            unique_trades.append(result_trade)
-
-        return unique_trades
+                spend_asset=from_asset,
+                spend_amount=from_amount,
+                receive_asset=to_asset,
+                receive_amount=to_amount,
+                fee_amount=fee_amount,  # type: ignore[arg-type]
+                fee_asset=fee_asset,  # type: ignore[arg-type]
+                event_identifier=f'{EVENT_IDENTIFIER_PREFIX}{hash_binance_csv_row(trade_rows[0])}',
+                spend_notes='Imported from binance CSV file. Binance operation: Buy / Sell',
```

Comment:
Question here: Will these be fixed during DB upgrade? You said they were imported completely wrong before. Couldn't you add some logic using the notes here to identify and fix?

Reply:
Isaac made a note for this here: https://github.com/orgs/rotki/projects/11/views/2?pane=issue&itemId=106596282